sql/core/src/test/scala/org/apache/spark/sql/connector/DataSourceV2SQLSuite.scala

```diff
@@ -3560,11 +3560,10 @@ class DataSourceV2SQLSuiteV1Filter
         val exception = analysisException(
           // Rand function is not foldable
           s"ALTER TABLE tab ADD COLUMN col2 DOUBLE DEFAULT rand()")
-        assert(exception.getSqlState == "42623")
-        assert(exception.errorClass.get == "INVALID_DEFAULT_VALUE.NOT_CONSTANT")
-        assert(exception.messageParameters("colName") == "`col2`")
-        assert(exception.messageParameters("defaultValue") == "rand()")
-        assert(exception.messageParameters("statement") == "ALTER TABLE")
+        assert(exception.getSqlState == "42K0E")
```

Comment:
Note: the exception changes here because now existsDefault is analyzed as part of the main loop, instead of separately by call to ResolveDefaultColumns.  However, as that code had custom pretty exceptions for ADD COLUMN DEFAULT VALUE (see https://github.com/apache/spark/pull/42810), those get lost.

I was able to fix the other ones, but this one is not able to as I dont have enough information at the end of Analyzer phase when value is copied to existsDefault.  Some of the expressions are replaced later by optimizer (ReplaceCurrentLike)